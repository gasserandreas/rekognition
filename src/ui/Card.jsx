import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { Box } from 'grommet';

import { Colors } from '../styles';

export const StyledCard = styled(Box)`
  background-color: ${Colors.ColorsPalette.White};
  border-radius: 3px;
  box-sizing: border-box;
  margin: 1rem 0;
`;

/** Simple card layout container. Add your children
 * and use default Grommet props to style this component */
const Card = ({ children, ...props }) => (
  <StyledCard
    elevation="small"
    pad="medium"
    {...props}
  >{children}</StyledCard>
);

Card.propTypes = {
  /** Specify child to be rendered */
  children: PropTypes.node.isRequired,
};

export default Card;
