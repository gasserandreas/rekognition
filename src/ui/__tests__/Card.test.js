import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Card, { StyledCard } from '../Card';

// test styling
it('Card should render correctly', () => {
  const wrapper = shallow(<Card><p>Hello</p></Card>);
  expect(toJson(wrapper.dive())).toMatchSnapshot();
});

it('Card styles should be consistent', () => {
  const wrapper = shallow(<StyledCard />);
  expect(toJson(wrapper)).toMatchSnapshot();
});