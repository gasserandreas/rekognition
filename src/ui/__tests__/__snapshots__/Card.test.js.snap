// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<div
  className="sc-bdVaJa fmAdHF StyledBox-sc-13pk1d4-0 gWJLey"
>
  <p>
    Hello
  </p>
</div>
`;

exports[`Card styles should be consistent 1`] = `
<div
  className="sc-bdVaJa fmAdHF StyledBox-sc-13pk1d4-0 IRSNj"
/>
`;
